cmd/ebrelayer/relayer/ethereum.go

```diff
@@ -228,6 +231,11 @@ func (sub EthereumSub) Start(completionEvent *sync.WaitGroup) {
 			// The user who starts this must provide a valid last processed block
 			if lastProcessedBlock.Cmp(big.NewInt(0)) == 0 {
 				lastProcessedBlock.Sub(endingBlock, big.NewInt(1))
+			} else if catchUpNeeded {
+				// if we need to catch up, then do that and let the relayer know that we don't need to catch up again
```

Comment:
The comment doesn't match the code - it says "then do that" but all we're doing is setting the bool.

Reply:
So it says not to mess with the lastProcessed block if it needs to catch up. Once that bool is set, the next time around, it will miss that statement and change the lastProcessedBlock value to the current block we are looking at on the line 2 lines down.